### ISSUE TITLE
[DataGrid] onMenuClose called too often

### ISSUE THREAD

**Issue by Janpot** [mui/mui-x#11336]
### Search keywords

onMenuClose

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: https://stackblitz.com/edit/react-uvdnfy?file=Demo.tsx

Steps:
1. Open the example
2. Open the stackblitz console
3. Click on a DataGrid item

### Current behavior

Prints `"menu closes"` about 10 times, while no menu was open or is closing.

Additionally, the jsodc annotation and the docs are quite unclear about what this property is for. Which menu?

### Expected behavior

The handler is not called when no menu is closing

The docs are clear about which menu closes and what this property could be used for.

### Context

Not trying to accomplish anything specific, was just looking for a event that fires when the filtermenu closes. Tried out this event to see what it does.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by DanailH** [comment]
I agree, the docs are not idea. Also, another weird thing that I noticed is that the callback is being called when you open the menu as well (try opening the filter pannel for example -> it prints the console log).
